code-llama-7b-instruct/model/model.py

```diff
@@ -0,0 +1,69 @@
+from transformers import AutoTokenizer, AutoModelForCausalLM
+import torch
+from typing import Dict
+
+DEFAULT_SYSTEM_PROMPT = """\
+You are a helpful, respectful and honest coding assistant. Always answer as helpfully as possible, while being safe.  Your answers should not include any harmful, unethical, racist, sexist, toxic, dangerous, or illegal content. Please ensure that your responses are socially unbiased and positive in nature.
+
+If a question does not make any sense, or is not factually coherent, explain why instead of answering something not correct. If you don't know the answer to a question, please don't share false information."""
+
+
+class Model:
+    MODEL_NAME = "codellama/CodeLlama-7b-Instruct-hf"
+
+    def __init__(self, data_dir: str, config: Dict, **kwargs):
+        self._data_dir = data_dir
+        self._config = config
+        self.model = None
+        self.tokenizer = None
+
+    def preprocess(self, request: dict):
+        generate_args = {
+            "max_new_tokens": 128,
```

Comment:
nit: could we have this be:
```
"max_new_tokens": request.get("max_tokens", 128)
```

instead? (and ditto for the other parameters)

Reply:
Good suggestion. I updated the code.